frontend/src/hooks/useApi.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { useEffect, useState } from "react";
+import axios, { AxiosError, AxiosResponse } from "axios";
+
+import { API_BASE_URL } from "@constants";
+import { useAppContext } from "@providers/ContextProvider";
+
+const errorHandler = (error: AxiosError): string => {
+  if (error.response) {
+    return `Server responded with error: ${error.response.status}`;
+  } else if (error.request) {
+    return `No response received: ${error.request}`;
+  } else {
+    return `Error occurred: ${error.message}`;
+  }
+};
+
+export const useApiGetRequest = <T,>(path: string, params?: any) => {
+  const [data, setData] = useState<T | null>(null);
+  const [isPending, setIsPending] = useState<boolean>(false);
+
+  const { setError } = useAppContext();
+
+  useEffect(() => {
```

Comment:
I'm not sure you need a useEffect here? This should most likely be a .ts file. It doesn't need to be tied to the React Lifecycle.